tests/sanity/extra/action-group.py

```diff
@@ -0,0 +1,127 @@
+        'exclusions': [],
+        'doc_fragment': 'community.general.proxmox.actiongroup_proxmox',
+    },
+}
+
+
+def main():
+    """Main entry point."""
+
+    # Load redirects
+    meta_runtime = 'meta/runtime.yml'
+    self_path = 'tests/sanity/extra/action-group.py'
+    try:
+        with open(meta_runtime, 'rb') as f:
+            data = yaml.safe_load(f)
+        action_groups = data['action_groups']
+    except Exception as exc:
+        print(f'{meta_runtime}: cannot load action group: {exc}')
+        return
+
+    for action_group in action_groups:
+        if action_group not in ACTION_GROUPS:
+            print(f'{meta_runtime}: found unknown action group {action_group!r}; likely {self_path} needs updating')
+    for action_group, action_group_data in list(ACTION_GROUPS.items()):
+        if action_group not in action_groups:
+            print(f'{meta_runtime}: cannot find action group {action_group!r}; likely {self_path} needs updating')
+
+    modules_directory = 'plugins/modules/'
+    modules_suffix = '.py'
+
+    for file in os.listdir(modules_directory):
+        if not file.endswith(modules_suffix):
+            continue
+        module_name = file[:-len(modules_suffix)]
+
+        for action_group, action_group_data in ACTION_GROUPS.items():
+            action_group_content = action_groups.get(action_group) or []
+            path = os.path.join(modules_directory, file)
+
+            if not action_group_data['pattern'].match(module_name):
+                if module_name in action_group_content:
+                    print(f'{path}: module is in action group {action_group!r} despite not matching its pattern as defined in {self_path}')
+                continue
+
+            should_be_in_action_group = module_name not in action_group_data['exclusions']
+
+            if should_be_in_action_group:
+                if module_name not in action_group_content:
+                    print(f'{meta_runtime}: module {module_name!r} is not part of {action_group!r} action group')
+                else:
+                    action_group_content.remove(module_name)
+
+            documentation = []
+            in_docs = False
+            with open(path, 'r', encoding='utf-8') as f:
+                for line in f:
+                    if line.startswith('DOCUMENTATION ='):
+                        in_docs = True
+                    elif line.startswith(("'''", '"""')) and in_docs:
+                        in_docs = False
+                    elif in_docs:
+                        documentation.append(line)
+            if in_docs:
+                print(f'{path}: cannot find DOCUMENTATION end' % (path))
+            if not documentation:
+                print(f'{path}: cannot find DOCUMENTATION' % (path))
+                continue
+
+            try:
+                docs = yaml.safe_load('\n'.join(documentation))
+                if not isinstance(docs, dict):
+                    raise Exception('is not a top-level dictionary')
```

Comment:
The test has to react on these situations anyway, so it can also report errors. It's better to check for an error more than once than to always assume that someone else is reporting it, and in the end nobody is...

Reply:
ok, fair enough